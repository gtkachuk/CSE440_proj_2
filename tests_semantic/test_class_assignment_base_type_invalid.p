program testClassAssignmentBaseTypeInvalid;

class BB

BEGIN
   VAR row : integer;
END

class AA extends BB

BEGIN
   VAR works : boolean;
END


class testClassAssignmentBaseTypeInvalid

BEGIN
   
   VAR objectA	: AA;
       objectB	: BB;

FUNCTION testClassAssignmentBaseTypeInvalid;
BEGIN
   objectB := objectA
END

END
.
