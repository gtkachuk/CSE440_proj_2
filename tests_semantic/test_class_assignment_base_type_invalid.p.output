ERROR:26:Type mismatch between "BB" and "AA"
Errors detected. Exiting.
