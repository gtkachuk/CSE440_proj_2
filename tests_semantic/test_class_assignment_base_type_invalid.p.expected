ERROR:26:Class "aa" is not a base class of "bb" in the assignment
Errors detected. Exiting.
